packages/graphql/tests/integration/directives/cypher/cypher-with-non-node-types.int.test.ts

```diff
@@ -0,0 +1,316 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TestHelper } from "../../../utils/tests-helper";
+
+describe("cypher directive targeting non node types", () => {
+    const testHelper = new TestHelper();
+
+    afterEach(async () => {
+        await testHelper.close();
+    });
+
+    test("field", async () => {
+        const Movie = testHelper.createUniqueType("Movie");
+        const Link = testHelper.createUniqueType("Link");
+
+        const typeDefs = /* GraphQL */ `
+            type  ${Movie} @node {
+               title: String
+               id: String!
+               link: ${Link}! @cypher(statement: """
+               MATCH(l:${Link})
+               WHERE l.movieId=this.id
+               RETURN l {.name, .url} as link
+               """, columnName: "link")
+            }
+
+            type ${Link} {
+                movieId: String!
+                url: String!
+                name: String!
+            }
```

Comment:
How about this kind of case where the Cypher might return a more complex object? 

```suggestion
            type ${Link} {
                movieId: String!
                url: String!
                name: String!
                somethingElse: SomethingElse!
            }
            
            type ${SomethingElse} {
               something: String!
            }
```

Reply:
I'll add the test, but at that point is just GraphQL parsing the return object